### PR TITLE
Release/7.6.0

### PR DESCRIPTION
**Pull request by cthomas-newrelic**
## New in this Release
* Adds Gradle Plugin Portal support
* Upgrades JDK support (to 22)
* Updates plugin _"max AGP version"_ (to 8.6)
* Add feature flag for event persistence

## Fixes:
- Fixed Crash appbuild and App Version Mismatch 
- Fixed Where Crashes are not ingested due to empty stacktraceElemets
- Fixed a NullValuepointer exception
- Fixed a minor param mismatch in gradle plugin
- Fixed mobile data payload not ingested

### CODE REVIEW

**Review by ndesai-newrelic — APPROVED** [review]

**Review by ndesai-newrelic — APPROVED** [review]

**Comment by ywang-nr** [comment]
Multiple tests failed, please fix.

**Review by ndesai-newrelic — APPROVED** [review]

File: plugins/gradle/build.gradle [hunk 1]
```diff
@@ -161,8 +166,27 @@ tasks.named('check') {
...
+    classifier = null
+}
+
 artifacts {
-    archives jar
+    archives shadowJar
+}
+
+gradlePlugin {
+    website = 'https://docs.newrelic.com/docs/mobile-monitoring/new-relic-mobile-android/get-started/introduction-new-relic-mobile-android/'
```
Comment (ywang-nr) [inline plugins/gradle/build.gradle:178]: Suspect that we need to add additional info here.
Sonatype close repo is failed due to error message:
Invalid POM: /com/newrelic/agent/android/gradle/7.6.0/gradle-7.6.0.pom: Project name missing, Project description missing, Project URL missing, License information missing, SCM URL missing, Developer information missing

Comment (cthomas-newrelic) [reply plugins/gradle/build.gradle:178]: This block is only for teh Gradle Plugin Portal. The error is from Sonatype, but other POM changes to get GPP to work are probably the cause.

File: agent-core/src/main/java/com/newrelic/agent/android/harvest/DataToken.java [hunk 2]
```diff
@@ -60,6 +60,13 @@ public boolean isValid() {
         return accountId > 0 && agentId > 0;
     }
 
+    public static DataToken newFromJson(JsonArray jsonArray) {
+        final DataToken dataToken = new DataToken();
+        dataToken.setAccountId(jsonArray.get(0).getAsInt());
```
Comment (cthomas-newrelic) [inline agent-core/src/main/java/com/newrelic/agent/android/harvest/DataToken.java:65]: Is jsonArray valid? Is there an element at position 0?

Comment (ndesai-newrelic) [reply agent-core/src/main/java/com/newrelic/agent/android/harvest/DataToken.java:65]: yes, we are only using it for reading data token previously stored crash in sharedpreference.

File: settings.gradle [hunk 3]
```diff
@@ -25,6 +26,7 @@ pluginManagement {
         id("java")
         id("maven-publish")
         id("io.github.gradle-nexus.publish-plugin") version "${versions.nexus}"
+//        id("com.gradle.plugin-publish") version "${versions.gpp}" apply false
         id("com.android.library") version "${versions.agp}"
```
Comment (ywang-nr) [inline settings.gradle:30]: revert this back

**Review by smalsam-newr — APPROVED** [review]

**Pull request merged**